runtime/v2/vm/src/main/java/com/walmartlabs/concord/svm/InMemoryState.java

```diff
@@ -83,9 +83,30 @@ public Frame peekFrame(ThreadId threadId) {
     }
 
     @Override
-    public void popFrame(ThreadId threadId) {
+    public void popFrame(ThreadId threadId, UnstoppableCommandHandler handler) {
         log.trace("popFrame {}", threadId);
 
+        Frame currentFrame = peekFrame(threadId);
+
+        // finalizers
+        Collection<Exception> causes = new ArrayList<>();
+
+        currentFrame.getCommands()
```

Comment:
We could store all the `finally` commands as a block command in a separate field in `Frame`, I think.
And fill it when we `build()` the frame. It should not be too different from the exceptionHandler there.

Reply:
will close this PR